protocol/str.go

```diff
@@ -0,0 +1,19 @@
+package protocol
+
+import "github.com/coniks-sys/coniks-go/merkletree"
+
+type DirSTR struct {
+	merkletree.SignedTreeRoot
```

Comment:
I think we can save memory by having one pointer to, say, the latest STR for all requests in the same epoch, instead of allocating new one for each request. Is that reasonable?